src/index.ts

```diff
@@ -43,97 +63,94 @@ export class WalletPluginAnchor implements WalletPlugin {
-            elements: [
-                {
-                    type: 'qr',
-                    data: String(request),
-                },
-                {
-                    type: 'button',
-                    label: 'Open Anchor',
-                    data: String(request),
-                },
-            ],
-        })
+    login(context: LoginContext): Promise<WalletPluginLoginResponse> {
+        return new Promise((resolve, reject) => {
+            context.ui?.status('Preparing request for Anchor...')
 
-        // Use the buoy-client to create a promise and wait for a response to the identity request
-        const walletResponse = receive({...callback, WebSocket})
-
-        // TODO: Implement cancel logic from the UI
-        const cancel = new Promise<never>(() =>
-            //resolve,
-            //reject
-            {
-                // // Code from anchor-link...
-                // t.onRequest(request, (reason) => {
-                //     if (done) {
-                //         // ignore any cancel calls once callbackResponse below has resolved
-                //         return
-                //     }
-                //     const error = typeof reason === 'string' ? new CancelError(reason) : reason
-                //     if (t.recoverError && t.recoverError(error, request) === true) {
-                //         // transport was able to recover from the error
-                //         return
-                //     }
-                //     walletResponse
-                //     callback.cancel()
-                //     reject(error)
-                // })
-            }
-        )
+            // Create the identity request to be presented to the user
+            createIdentityRequest(context)
+                .then(({callback, request, requestKey, privateKey}) => {
+                    // Tell Wharf we need to prompt the user with a QR code and a button
+                    const promptResonse = context.ui?.prompt({
+                        title: 'Login with Anchor',
+                        body: 'Scan the QR-code with Anchor on another device or use the button to open it here.',
+                        elements: [
+                            {
+                                type: 'qr',
+                                data: String(request),
+                            },
+                            {
+                                type: 'button',
+                                label: 'Open Anchor',
+                                data: String(request),
+                            },
+                        ],
+                    })
 
-        // Await a promise race to wait for either the wallet response or the cancel
-        const callbackResponse = await Promise.race([walletResponse, cancel])
+                    promptResonse
+                        .catch((error) => {
+                            reject(error)
+                        })
```

Comment:
If you close the QR/click prompt, it moves on to the next step here.

This code needs to include something like this: 

https://github.com/wharfkit/transact-plugin-autocorrect/blob/master/src/index.ts#L227-L229